stdlib/experimental/set.go

```diff
@@ -0,0 +1,210 @@
+	"github.com/influxdata/flux/plan"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+)
+
+const SetKind = "setExperimental"
+
+type SetOpSpec struct {
+	Object values.Object `json:"object"`
+}
+
+func init() {
+	setSignature := flux.FunctionSignature(
+		map[string]semantic.PolyType{
+			"o": semantic.Tvar(1),
+		},
+		[]string{"o"},
+	)
+
+	flux.RegisterPackageValue("experimental", "set", flux.FunctionValue(SetKind, createSetOpSpec, setSignature))
+	flux.RegisterOpSpec(SetKind, newSetOp)
+	plan.RegisterProcedureSpec(SetKind, newSetProcedure, SetKind)
+	execute.RegisterTransformation(SetKind, createSetTransformation)
+}
+
+func createSetOpSpec(args flux.Arguments, a *flux.Administration) (flux.OperationSpec, error) {
+	if err := a.AddParentFromArgs(args); err != nil {
+		return nil, err
+	}
+
+	spec := new(SetOpSpec)
+	o, err := args.GetRequiredObject("o")
+	if err != nil {
+		return nil, err
+	}
+	spec.Object = o
+	return spec, nil
+}
+
+func newSetOp() flux.OperationSpec {
+	return new(SetOpSpec)
+}
+
+func (s *SetOpSpec) Kind() flux.OperationKind {
+	return SetKind
+}
+
+type SetProcedureSpec struct {
+	plan.DefaultCost
+	Object values.Object
+}
+
+func newSetProcedure(qs flux.OperationSpec, pa plan.Administration) (plan.ProcedureSpec, error) {
+	s, ok := qs.(*SetOpSpec)
+	if !ok {
+		return nil, fmt.Errorf("invalid spec type %T", qs)
+	}
+	p := &SetProcedureSpec{
+		Object: s.Object,
+	}
+	return p, nil
+}
+
+func (s *SetProcedureSpec) Kind() plan.ProcedureKind {
+	return SetKind
+}
+func (s *SetProcedureSpec) Copy() plan.ProcedureSpec {
+	ns := new(SetProcedureSpec)
+	ns.Object = values.NewObjectWithBacking(s.Object.Len())
+	s.Object.Range(func(k string, v values.Value) {
+		ns.Object.Set(k, v)
+	})
+	return ns
+}
+
+func createSetTransformation(id execute.DatasetID, mode execute.AccumulationMode, spec plan.ProcedureSpec, a execute.Administration) (execute.Transformation, execute.Dataset, error) {
+	s, ok := spec.(*SetProcedureSpec)
+	if !ok {
+		return nil, nil, fmt.Errorf("invalid spec type %T", spec)
+	}
+	cache := execute.NewTableBuilderCache(a.Allocator())
+	d := execute.NewDataset(id, mode, cache)
+	t := NewSetTransformation(d, cache, s)
+	return t, d, nil
+}
+
+type setTransformation struct {
+	d     execute.Dataset
+	cache execute.TableBuilderCache
+
+	object values.Object
+}
+
+func NewSetTransformation(
+	d execute.Dataset,
+	cache execute.TableBuilderCache,
+	spec *SetProcedureSpec,
+) execute.Transformation {
+	return &setTransformation{
+		d:      d,
+		cache:  cache,
+		object: spec.Object,
+	}
+}
+
+func (t *setTransformation) RetractTable(id execute.DatasetID, key flux.GroupKey) error {
+	// TODO
+	return nil
+}
+
+func (t *setTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
+	key := tbl.Key()
+	updateKey := false
+	t.object.Range(func(k string, _ values.Value) {
+		updateKey = updateKey || execute.HasCol(k, key.Cols())
+	})
+	if updateKey {
```

Comment:
Is this necessary? Would it possible for `set` to just extend and not to update silently?

Reply:
I am not sure I follow, if the set object is going to overwrite existing columns in the group key then it  needs to update the key.

Reply:
I'm saying that we should have a separate function, call it `update` maybe, for explicitly updating columns that already exist on the input tables. Then `set` would only be used to add columns that don't exist on the input tables.